nbody_gravitational/NBodyGravitational.jl

```diff
@@ -0,0 +1,173 @@
+module NBodyGravitational
+
+using DiffEqBase, DifferentialEquations, StaticArrays, Plots, StaticArrays
+
+
+grav_const = 6.673e-11
+
+#=
+    Represents a body/particle in an N-body gravitational problem
+    m - mass
+    r - initial position
+    v - initial velocity
+=#
+struct MassBody{mType, cType}
+    m :: mType
+    r :: Vector{cType}
+    v :: Vector{cType}
+    
+    function MassBody{mType,cType}(m::mType, r::Vector{cType}, v::Vector{cType})
+        @assert length(r)==length(v) "The length of the position (r) and velocity (v) components should match"
```

Comment:
If you do a static array then this can be type checked for free

Reply:
Ok